internal/test/log.go

```diff
@@ -0,0 +1,47 @@
+// Copied from "github.com/go-logr/logr/testing"
+// With added implementation of log.Info, log.V, log.WithName and log.WithValues
+
+package test
+
+import (
+	"testing"
+
+	"github.com/go-logr/logr"
+)
+
+// TestLogger is a logr.Logger that prints through a testing.T object.
+type TestLogger struct {
+	T      *testing.T
+	v      int
+	name   string
+	values []interface{}
+}
+
+var _ logr.Logger = TestLogger{}
```

Comment:
dont think we need to force the inheritance 

```suggestion
```

Reply:
This is just a way of checking that the TestLogger provides the logr.Logger interface. 
It is copied from the github.com/go-logr/logr/testing package, so I'll leave it for now.